### ISSUE TITLE
Embedding metadata is not read or displayed in TensorBoard 2.3.0

### ISSUE THREAD

**Issue by toddwasson** [tensorflow/tensorboard#4246]
Hello, I have trained a model (with TensorFlow 2.3.1) including embeddings via `keras.layers.Embedding` and prepared label files per embedding dictionary, which I point to via the `embeddings_metadata` parameter of `keras.callbacks.TensorBoard`.  TensorBoard executes and displays the various embedding vectors in the Projector without issue, but the embeddings are not labeled per the label files.  The label files are successfully created and stored in the log directory, and I can use them manually just fine via the `Load` button in the TensorBoard interface.

I debugged the `projector_plugin.py` code and determined that `_get_metadata_file_for_tensor()` is never called (verified via `prints`, debugging with `pdb`, etc.).  This method looks to be the only place in the code used to retrieve the `metadata_path` from which the labels would be read, and is apparently only accessed via the `/metadata` HTTP route.

Incidentally when I add the various `Embedding` layers in Keras, the first receives the `tensor_name` `layer_with_weights-0/embeddings/.ATTRIBUTES/VARIABLE_VALUE` but the subsequent ones are `layer_with_weights-1`, `2`, etc.  I see the `keras.callbacks.TensorBoard` hard-codes them to `layer_with_weights-0` but I think this is a separate issue as I have manually corrected the `projector_config.pbtxt` file and, as I mentioned above, the `_get_metadata_file_for_tensor()` method is never called anyway so they aren't even trying to be parsed and used to label tensors.

Any thoughts?  Am I missing something?

Thanks!

Todd

**Comment by psybuzz** [comment]
cc @hfiller

**Comment by psybuzz** [comment]
Thanks for the report!  I was able to repro this with the following setup:

In `root/main.py`, train a model with a Keras embedding layer, using
`tf.keras.callbacks.TensorBoard(log_dir='./logs', embeddings_metadata="my_labels.tsv", ...)`

Keras produced the following directory structure:
```
logs/
  - projector_config.pbtxt
  - train/
    - keras_embedding.ckpt-0.index
  - validation/
```

where the projector_config.pbtxt contains
```
embeddings {
  tensor_name: "layer_with_weights-0/embeddings/.ATTRIBUTES/VARIABLE_VALUE"
  metadata_path: "my_labels.tsv"
}
```

The Projector seems to get confused, and looks for the config in `logs/train/projector_config.pbtxt`, when the actual file exists at `logs/projector_config.pbtxt`.  I got TensorBoard to properly slurp the config file by moving `logs/projector_config.pbtxt` to `logs/train/projector_config.pbtxt`, and ensuring that the `metadata_path: ...` field in the pbtxt properly points to the labels file.

Ideally this would be fixed in TensorBoard / Keras.  As a workaround for now, could you please try moving your `projector_config.pbtxt` into the same directory as your Keras checkpoint, update the `metadata_path`, and see if TB automatically loads your labels?

**Comment by toddwasson** [comment]
Great news: that fix works for a `keras.layers.Embedding` at `layer_with_weights-0`!  Thank you!

Less great news: it still doesn't work for `keras.layers.Embedding` layers at `layer_with_weights-1` and beyond -- in fact, it stores them all as `layer_with_weights-0` at saving time.  (The fix might actually work if I manually update the `logs/train/projector_config.pbtxt` to correctly point to each tensor.)  However this is actually a different issue, and it's also possible that I'm doing something wrong, so if you'd like me to open a separate issue for that I'd be happy to do so.

Either way, thank you for chasing this down.  It provides me a path forward in the meantime and hopefully will be easy to fix upstream.

**Comment by rmothukuru** [comment]
@toddwasson,
Closing this issue as it is resolved. Please raise a new issue for the different problem that you are facing. Thanks!
